glib/src/variant_dict.rs

```diff
@@ -59,6 +59,20 @@ impl VariantDict {
         }
     }
 
+    /// Look up a typed value from this `VariantDict`.
+    ///
+    /// The given `key` is looked up in `self`.
+    ///
+    /// This will return `None` if the `key` is not present in the dictionary
+    /// or if it is present but the type of the value does not match the
+    /// expected type.
```

Comment:
Should non-presence maybe be distinguished in the return type from presence-but-wrong-type? Using a `Result<Option<T>, WrongTypeError>` probably will make it harder to use?

Reply:
I think returning a `Result<Option<T>>` makes more sense though, otherwise we can't know what failed.

Reply:
`headers::HeaderMapExt` has `typed_get()` and `typed_try_get()`, the latter returning a `Result` and the former flattening into a single `Option`.

Reply:
Should we provide both then?

Reply:
https://docs.rs/headers/0.3.4/headers/trait.HeaderMapExt.html#tymethod.typed_get for reference

Reply:
I made this method mirroring the existing `Variant::get()`.  Should we also add `try_get()` then?

Reply:
`Variant::get()` can only fail if the type doesn't match, but this here would fail if either that happens or the given key is not stored in the dict without the caller being able to distinguish the two cases.

Arguably, `Variant::get()` should also return a `Result` as that's an actual error case and not just a non-existing value.

Reply:
Right.   I was thinking to add `Variant::try_get()` because in my code I usually end up converting `Option` :arrow_right: `Result` for that case too; see e.g. https://github.com/coreos/rpm-ostree/pull/3034/files#diff-e618d2cef0ae2cb833c18e5603d050edbf916eeac5b66392487572aab2d8d32cR114

Reply:
If it was a `Result<T, glib::BoolError>` or something else generic that wouldn't probably help you much, you'd still need a `map_err()` in your code I assume?

Reply:
(But yes that should really return a `Result` IMHO)

Reply:
OK, updated!

Reply:
> If it was a Result<T, glib::BoolError> or something else generic that wouldn't probably help you much, you'd still need a map_err() in your code I assume?

That's a good point.  And I also notice we were missing the `impl Error`, fixed that.